src/components/Field.vue

```diff
@@ -429,15 +434,21 @@ export default {
       let onEvent;
 
       if (this.onEvent) {
-        onEvent = new Function(this.onEvent)();
+        onEvent = typeof this.onEvent === 'function'
+          ? this.onEvent
+          : new Function(this.onEvent)();
       }
 
-      this.provider = Fliplet.Widget.open(this.package, {
-        selector: target ? target[0] : undefined,
-        data: typeof value === 'object'
+      if (!('data' in this)) {
+        data = typeof value === 'object'
           // Normalize Vue objects into plain JSON objects
           ? JSON.parse(JSON.stringify(value))
-          : value,
+          : value;
+      }
+
+      this.provider = Fliplet.Widget.open(this.package, {
+        selector: target ? target[0] : undefined,
+        data,
         onEvent: onEvent
```

Comment:
[MINOR] @tonytlwu, for sake of consistency

```suggestion
        onEvent
```

Reply:
Update commited